tools/testgrid-analysis/cmd/data.go

```diff
@@ -0,0 +1,146 @@
+	// Fetch test data
+	rowsURL := fmt.Sprintf("http://testgrid-data.k8s.io/api/v1/dashboards/%s/tabs/%s/rows", dashboard, tab)
+	headersURL := fmt.Sprintf("http://testgrid-data.k8s.io/api/v1/dashboards/%s/tabs/%s/headers", dashboard, tab)
+
+	var testData apipb.ListRowsResponse
+	var headerData apipb.ListHeadersResponse
+	protojson.Unmarshal(fetchJson(rowsURL), &testData)
+	protojson.Unmarshal(fetchJson(headersURL), &headerData)
+
+	var allTests []string
+	for _, row := range testData.Rows {
+		allTests = append(allTests, row.Name)
+	}
+
+	summaries := []*TestResultSummary{}
+	// Process rows
+	for _, row := range testData.Rows {
+		t := processRow(dashboard, tab, row, allTests, headerData.Headers)
+		summaries = append(summaries, t)
+	}
+	return summaries
+}
+
+func processRow(dashboard, tab string, row *apipb.ListRowsResponse_Row, allTests []string, headers []*apipb.ListHeadersResponse_Header) *TestResultSummary {
+	t := TestResultSummary{Name: shortenTestName(row.Name), FullName: row.Name}
+	// we do not want to create issues for a parent test.
+	if isParentTest(row.Name, allTests) {
+		return &t
+	}
+	if !strings.HasPrefix(row.Name, "go.etcd.io") {
+		return &t
+	}
+	total := 0
+	failed := 0
+	logs := []string{}
+	for i, cell := range row.Cells {
+		// ignore tests with status not in the validTestStatuses
+		// cell result codes are listed in https://github.com/GoogleCloudPlatform/testgrid/blob/main/pb/test_status/test_status.proto
+		if _, ok := validTestStatusesInt[cell.Result]; !ok {
+			if cell.Result != 0 {
+				skippedTestStatuses[cell.Result] = struct{}{}
+			}
+			continue
+		}
+		total += 1
+		if _, ok := failureTestStatusesInt[cell.Result]; ok {
+			failed += 1
+			header := headers[i]
+			// markdown table format of | commit | log |
+			logs = append(logs, fmt.Sprintf("| %s | %s | https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/%s/%s |", strings.Join(header.Extra, ","), header.Started.AsTime().String(), tab, header.Build))
+		}
+	}
+	t.FailedRuns = failed
+	t.TotalRuns = total
+	t.FailureLogs = logs
+	t.FailureRate = float32(failed) / float32(total)
+	if t.FailedRuns > 0 {
+		dashboardUrl := fmt.Sprintf("[%s](https://testgrid.k8s.io/%s#%s)", tab, dashboard, tab)
+		t.IssueBody = fmt.Sprintf("## %s Test: %s \nTest failed %.1f%% (%d/%d) of the time\n\nfailure logs are:\n| commit | started | log |\n| --- | --- | --- |\n%s\n",
+			dashboardUrl, t.FullName, t.FailureRate*100, t.FailedRuns, t.TotalRuns, strings.Join(t.FailureLogs, "\n"))
+		t.IssueBody += "\nPlease follow the [instructions in the contributing guide](https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#check-for-flaky-tests) to reproduce the issue.\n"
+	}
+	return &t
+}
+
+// isParentTest checks if a test is a rollup of some child tests.
+func isParentTest(test string, allTests []string) bool {
+	for _, t := range allTests {
+		if t != test && strings.HasPrefix(t, test+"/") {
+			return true
+		}
+	}
+	return false
+}
+
+func fetchJson(url string) []byte {
```

Comment:
I'm currently working on enabling the `var-naming` linter rule, and it will complain about this function. It should be named `fetchJSON(...)` :sweat_smile: